src/components/common/AddressTextWithBlockExplorerModal.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import { createYesNoModal, showModal } from 'edge-components'
+import React from 'react'
+import type { Node } from 'react'
+import { Alert, Linking, TouchableOpacity } from 'react-native'
+import { sprintf } from 'sprintf-js'
+
+import * as Constants from '../../constants/indexConstants'
+import s from '../../locales/strings.js'
+import { Icon } from '../../modules/UI/components/Icon/Icon.ui.js'
+
+type AddressTextWithBlockExplorerModalProps = {
+  address: string,
+  addressExplorer: string | null,
+  children: Node
+}
+
+const AddressTextWithBlockExplorerModal = (props: AddressTextWithBlockExplorerModalProps) => {
+  const { address, addressExplorer, children } = props
+  const addressModal = async () => {
+    if (addressExplorer) {
+      const modal = createYesNoModal({
+        title: s.strings.modal_addressexplorer_message,
+        message: address,
+        icon: <Icon type={Constants.ION_ICONS} name={Constants.WALLET_ICON} size={30} />,
+        noButtonText: s.strings.string_cancel_cap,
+        yesButtonText: s.strings.string_ok_cap
+      })
+      return (await showModal(modal)) ? Linking.openURL(sprintf(addressExplorer, address)) : null
+    }
+    return Alert.alert(s.strings.modal_addressexplorer_null)
+  }
+  return <TouchableOpacity onPress={addressModal}>{children}</TouchableOpacity>
+}
+
+export default AddressTextWithBlockExplorerModal
```

Comment:
Do not use default exports. Always export/import the exact name of the object

Reply:
Noted